equinix/resource_network_device.go

```diff
@@ -991,6 +1169,15 @@ func updateNetworkDeviceResource(primary *ne.Device, secondary *ne.Device, d *sc
 			return fmt.Errorf("error reading Secondary: %s", err)
 		}
 	}
+	if primary.ClusterDetails != nil {
+		if v, ok := d.GetOk(networkDeviceSchemaNames["ClusterDetails"]); ok {
+			clusterDetailsFromSchema := expandNetworkDeviceClusterDetails(v.([]interface{}))
+			primary.ClusterDetails.ClusterNodeDetails = clusterDetailsFromSchema.ClusterNodeDetails
+		}
+		if err := d.Set(networkDeviceSchemaNames["ClusterDetails"], flattenNetworkDeviceClusterDetails(primary.ClusterDetails)); err != nil {
```

Comment:
To combine the Node and ClusterNodeDetails, we'll have to merge `primary.ClusterDetails` and the `nodes` response fields.